private/pkg/transport/http/httpserver/headers.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpserver
```

Comment:
Note this file is temporary.  It is only needed to maintain backwards compatibility with the CLI until `core` is fully updated to remove all usage of headers in context.

Having this here just temporarily allows us to release the CLI without breaking the BSR.

Reply:
Okay, I assume this is a straight copy of the existing `rpc`.

To confirm, 

1. merge this PR
2. update core to grab the headers directly within a connect interceptor
3. open another PR against buf to remove this

Reply:
I don't understand why this would affect the BSR at all - being on a context or not is a Golang-level concern, not a wire concern.